Doc/c-api/exceptions.rst

```diff
@@ -459,15 +463,17 @@ Querying the error indicator
             PyErr_Restore(type, value, traceback);
          }
 
-   .. deprecated:: 3.12
-
 
 .. c:function:: void PyErr_Restore(PyObject *type, PyObject *value, PyObject *traceback)
 
-    As of 3.12, this function is deprecated. Use :c:func:`PyErr_SetRaisedException` instead.
+   .. deprecated:: 3.12
+
+      Use :c:func:`PyErr_SetRaisedException` instead.
 
-   Set the error indicator from the three objects.  If the error indicator is
-   already set, it is cleared first.  If the objects are ``NULL``, the error
+   Set the error indicator from the three objects,
+   *type*, *value*, and *traceback*,
+   clearing the existing exception if one is set.
+   If the objects are ``NULL``, the error
    indicator is cleared.  Do not pass a ``NULL`` type and non-``NULL`` value or
    traceback.  The exception type should be a class.  Do not pass an invalid
    exception type or value. (Violating these rules will cause subtle problems
```

Comment:
I guess I was thinking of also revising this part along the lines of the previous, and programmatically marking the function as stealing a reference (if it isn't already), and perhaps with the warning message as in the previous. But up to you if you think it's worth it.

Reply:
Sure, sounds good to me.

Reply:
Hm, I'm afraid we'll have to reword most of `PyErr_Fetch` and `PyErr_Restore` in this case. Let's land this so we can close the linked issue.

Further improvements and amendments can be made in small increments, using the _skip issue_ label.